pyartcd/pyartcd/pipelines/operator_sdk_sync.py

```diff
@@ -0,0 +1,148 @@
+
+            et_builds = advisory.errata_builds
+            sdk_build = [i for i in et_builds[f'OSE-{self.version}-RHEL-8'] if re.search("openshift-enterprise-operator-sdk-container*", i)]
+            if not sdk_build:
+                self._logger.info("No SDK build to ship, update subtask 8 then close ...")
+                self._update_jira(self.parent_jira_key, 7, f"No SDK build to ship, operator_sdk_sync job: {os.environ.get('BUILD_URL')}")
+                return
+            build = koji.ClientSession(constants.BREW_SERVER).getBuild(sdk_build[0])
+        elif self.nvr:
+            build = koji.ClientSession(constants.BREW_SERVER).getBuild(self.nvr)
+        else:
+            self._logger.info("no assembly or nvr provided")
+            return
+
+        archlist = ["amd64", "arm64", "ppc64le", "s390x"]
+        sdkVersion = self._get_sdkversion(build['extra']['image']['index']['pull'][0])
+        self._logger.info(sdkVersion)
+        for arch in archlist:
+            self._extract_binaries(arch, sdkVersion, build['extra']['image']['index']['pull'][0])
+        if self.assembly:
+            self._update_jira(self.parent_jira_key, 7, f"operator_sdk_sync job: {os.environ.get('BUILD_URL')}")
+
+    def get_ad_jira_key(self, assembly, release_yaml):
+        if 'group' in release_yaml['releases'][assembly]['assembly'].keys():
+            extra_ad_id = release_yaml['releases'][assembly]['assembly']['group']['advisories']['extras']
+            parent_jira_key = release_yaml['releases'][assembly]['assembly']['group']['release_jira']
+            return extra_ad_id, parent_jira_key
+        if 'assembly' not in release_yaml['releases'][assembly]['assembly']['basis'].keys():
+            raise ValueError("Can not find jira and advisory number from assembly")
+        return self.get_ad_jira_key(release_yaml['releases'][assembly]['assembly']['basis']['assembly'], release_yaml)
+
+    def _get_sdkversion(self, build):
+        output = subprocess.getoutput(f"oc image info --filter-by-os amd64 -o json {build} | jq .digest")
+        shasum = re.findall("sha256:\\w*", output)[0]
+        cmd = f"oc image extract {constants.OPERATOR_URL}@{shasum} --path /usr/local/bin/{self.sdk}:. --confirm && chmod +x {self.sdk} && ./{self.sdk} version && rm {self.sdk}"
+        self._logger.info(cmd)
+        sdkvalue = subprocess.getoutput(cmd)
+        sdkversion = re.findall("v\\d.*-ocp", sdkvalue)[0]
+        return sdkversion
+
+    def _extract_binaries(self, arch, sdkVersion, build):
+        output = subprocess.getoutput(f"oc image info --filter-by-os {arch} -o json {build} | jq .digest")
+        shasum = re.findall("sha256:\\w*", output)[0]
+
+        rarch = arch
+        rarch = 'x86_64' if arch == 'amd64' else rarch
+        rarch = 'aarch64' if arch == 'arm64' else rarch
+        tarballFilename = f"{self.sdk}-{sdkVersion}-linux-{rarch}.tar.gz"
+
+        cmd = f"rm -rf ./{rarch} && mkdir ./{rarch}" + \
```

Comment:
If the shell script is long, I would recommend to keep the whole script in a multi-line string:
cmd = """#!/bin/bash
# some command
"""
or have the script stored in a separate file.